parity/main.rs

```diff
@@ -195,38 +241,70 @@ macro_rules! trace_main {
 }
 
 fn main() {
-	panic_hook::set();
-
-	// assuming the user is not running with `--force-direct`, then:
-	// if argv[0] == "parity" and this executable != ~/.parity-updates/parity, run that instead.
+	// the user has specified to run its originally installed binary (not via `parity-updater`)
 	let force_direct = std::env::args().any(|arg| arg == "--force-direct");
-	let exe = std::env::current_exe().ok();
-	let development = exe.as_ref().and_then(|p| p.parent().and_then(|p| p.parent()).and_then(|p| p.file_name()).map(|n| n == "target")).unwrap_or(false);
-	let same_name = exe.as_ref().map(|p| p.file_stem().map_or(false, |s| s == "parity") && p.extension().map_or(true, |x| x == "exe")).unwrap_or(false);
-	trace_main!("Starting up {} (force-direct: {}, development: {}, same-name: {})", std::env::current_exe().map(|x| format!("{}", x.display())).unwrap_or("<unknown>".to_owned()), force_direct, development, same_name);
+	
+	// absolute path to the current `binary`
+	let exe_path = std::env::current_exe().ok();
```

Comment:
This `"shouldn't"` fail it's a wrapper on top of the underlying OS.
Do you want unwrap/expect here? 

However, it's safe because if this fails we `fall` back to the locally installed version (or in other words don't spawn a child process)